### PR TITLE
libdb/5.3.28 - check for clang compiler alongside apple-clang

### PR DESCRIPTION
**Pull request by vlebourl**
Specify library name and version:  **libdb/5.3.28**

This recipe doesn't build on macos if the profile is using a custom clang version because line 131 is currently not applied. Checking whether compiler is in `["apple-clang", "clang"]` rather than only `"apple-clang"` fixes the issue.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
please comment in the title the recipe this is impacting :)

**Comment by prince-chrismc** [comment]
You will need to update the imports of this recipe for the changes to be merged https://github.com/conan-io/conan-center-index/pull/16669/files#diff-b153d74e53befdfc8efd529e900c2626fc8949f9453e701118ad5ba7fdcf2f12R1

File: recipes/libdb/all/conanfile.py [hunk 1]
```diff
@@ -145,13 +147,13 @@ def _configure_autotools(self):
         else:
             conf_args.extend(["--disable-shared", "--enable-static"])
         if self.options.with_tcl:
-            conf_args.append("--with-tcl={}".format(tools.unix_path(os.path.join(self.deps_cpp_info["tcl"].rootpath, "lib"))))
+            conf_args.append("--with-tcl={}".format(unix_path(os.path.join(self.deps_cpp_info["tcl"].rootpath, "lib"))))
```
Comment (prince-chrismc) [inline recipes/libdb/all/conanfile.py:150]:
```suggestion
            conf_args.append("--with-tcl={}".format(unix_path(self, os.path.join(self.deps_cpp_info["tcl"].rootpath, "lib"))))
```

File: recipes/libdb/all/conanfile.py [hunk 2]
```diff
@@ -61,63 +66,60 @@ def config_options(self):
...
-                shutil.copy(self._user_info_build["gnu-config"].CONFIG_SUB,
-                            os.path.join(self._source_subfolder, subdir, "config.sub"))
-                shutil.copy(self._user_info_build["gnu-config"].CONFIG_GUESS,
-                            os.path.join(self._source_subfolder, subdir, "config.guess"))
+                for gnu_config in [
+                    self.conf.get("user.gnu-config:config_guess", check_type=str),
+                    self.conf.get("user.gnu-config:config_sub", check_type=str),
+                ]:
+                    if gnu_config:
+                        copy(self, os.path.basename(gnu_config), src=os.path.dirname(gnu_config), dst=os.path.join(self.source_folder, self._source_subfolder, subdir))
```
Comment (prince-chrismc) [inline recipes/libdb/all/conanfile.py:113]:
```suggestion
                        copy(self, os.path.basename(gnu_config), src=os.path.dirname(gnu_config), dst=os.path.join(self._source_subfolder, subdir))
```

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**